src/web/components/Core/KeySecretReveal.tsx

```diff
@@ -0,0 +1,113 @@
+import { useEffect, useState } from 'react';
+
+import Popover from './Popover';
+import { StatusPopup } from './StatusPopup';
+
+import './KeySecretReveal.scss';
+
+const MAX_SHOWN_VALUE_LENGTH = 20;
+
+function ViewKeyButton({
+  value,
+  title,
+  confirmCopiedSecret,
+}: {
+  value: string;
+  title: string;
+  confirmCopiedSecret: () => void;
+}) {
+  const display =
+    value.length > MAX_SHOWN_VALUE_LENGTH
+      ? `${value.substring(0, MAX_SHOWN_VALUE_LENGTH / 2)}......${value.substring(
+          value.length - MAX_SHOWN_VALUE_LENGTH / 2,
+          value.length
+        )}`
+      : value;
+
+  return (
+    <>
+      {display}
+      {value.length > MAX_SHOWN_VALUE_LENGTH && (
+        <>
+          <Popover
+            triggerButton={
+              <button
+                className='icon-button show-button'
+                aria-label={title}
+                type='button'
+                title={title}
+                onClick={confirmCopiedSecret}
+              >
+                View
+              </button>
+            }
+          >
+            <p>{value}</p>
+          </Popover>
+          |
```

Comment:
I think we should remove the separator from this component so there are more loosely-coupled.